src/query/storage/m3/multi_fetch_result_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3x/ident"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	common       = "common"
+	short        = "short"
+	long         = "long"
+	unaggregated = "unaggregated"
+)
+
+// NB: Each seriesIterators has two seriesIterator; one with a constant ID which
+// will be overwritten as necessary by multi_fetch_result, and one per namespace
+// which should not be overwritten and should appear in the results.
+func generateSeriesIterators(
+	ctrl *gomock.Controller, ns string) encoding.SeriesIterators {
```

Comment:
Not sure, but would it be worth doing a fake implementation of `SeriesIterators` instead? This is kind of a long mock sequence, and it's a little less clear what the behavior is than a fake implementation.

Reply:
Hey, that's fair; unfortunately there's a lot of stuff we'd have to fake here (faked `SeriesIterator` means that we need to return faked `TagIterators` etc, etc, which is quite annoying

Perhaps if we'd had more tightly scoped iterators or something initially we'd be better off for the moment, but looks like we may be stuck with it as is unfortunately :(